0.12.0 undefined reference pthreads

When compiling 0.12.0 for openSUSE tumbleweed I get this error (see bottom).

My fast fix for this is applying this patch: http://pastie.org/10789411

> CMakeFiles/cppnetlib-client-connections.dir/client.cpp.o: In function `asio::detail::posix_global_impl<asio::system_executor::context_impl>::~posix_global_impl()':
> [   43s] client.cpp:(.text._ZN4asio6detail17posix_global_implINS_15system_executor12context_implEED2Ev[_ZN4asio6detail17posix_global_implINS_15system_executor12context_implEED5Ev]+0xdb): undefined reference to `pthread_join'
> [   43s] client.cpp:(.text._ZN4asio6detail17posix_global_implINS_15system_executor12context_implEED2Ev[_ZN4asio6detail17posix_global_implINS_15system_executor12context_implEED5Ev]+0xf9): undefined reference to `pthread_detach'
> [   43s] collect2: error: ld returned 1 exit status
> [   43s] libs/network/src/CMakeFiles/cppnetlib-client-connections.dir/build.make:99: recipe for target 'libs/network/src/libcppnetlib-client-connections.so.0.12.0' failed
> [   43s] make[2]: *** [libs/network/src/libcppnetlib-client-connections.so.0.12.0] Error 1
> [   43s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/cpp-netlib-0.12.0-final/build'
> [   43s] CMakeFiles/Makefile2:129: recipe for target 'libs/network/src/CMakeFiles/cppnetlib-client-connections.dir/all' failed
> [   43s] make[1]: *** [libs/network/src/CMakeFiles/cppnetlib-client-connections.dir/all] Error 2
> [   43s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/cpp-netlib-0.12.0-final/build'
> [   43s] Makefile:141: recipe for target 'all' failed
> [   43s] make: *** [all] Error 2


Cmake config output:

> [   19s] -- The C compiler identification is GNU 5.3.1
> [   19s] -- The CXX compiler identification is GNU 5.3.1
> [   19s] -- Check for working C compiler: /usr/bin/cc
> [   19s] -- Check for working C compiler: /usr/bin/cc -- works
> [   19s] -- Detecting C compiler ABI info
> [   19s] -- Detecting C compiler ABI info - done
> [   19s] -- Detecting C compile features
> [   19s] -- Detecting C compile features - done
> [   19s] -- Check for working CXX compiler: /usr/bin/c++
> [   19s] -- Check for working CXX compiler: /usr/bin/c++ -- works
> [   19s] -- Detecting CXX compiler ABI info
> [   20s] -- Detecting CXX compiler ABI info - done
> [   20s] -- Detecting CXX compile features
> [   20s] -- Detecting CXX compile features - done
> [   20s] -- Linking boost testing libs dynamically...
> [   20s] -- Boost version: 1.60.0
> [   20s] -- Found OpenSSL: /usr/lib64/libssl.so;/usr/lib64/libcrypto.so (found version "1.0.2g") 
> [   20s] -- Looking for pthread.h
> [   20s] -- Looking for pthread.h - found
> [   20s] -- Looking for pthread_create
> [   20s] -- Looking for pthread_create - not found
> [   20s] -- Looking for pthread_create in pthreads
> [   20s] -- Looking for pthread_create in pthreads - not found
> [   20s] -- Looking for pthread_create in pthread
> [   20s] -- Looking for pthread_create in pthread - found
> [   20s] -- Found Threads: TRUE  
> [   20s] -- Found PythonInterp: /usr/bin/python (found version "2.7.10") 
> [   20s] -- Configuring done
> [   20s] -- Generating done

--- a/libs/network/src/CMakeLists.txt
+++ b/libs/network/src/CMakeLists.txt
@@ -42,6 +42,7 @@ set_target_properties(cppnetlib-client-connections
     PROPERTIES VERSION ${CPPNETLIB_VERSION_STRING}
     SOVERSION ${CPPNETLIB_VERSION_MAJOR}
     PUBLIC_HEADER "${CPP-NETLIB_HEADERS}")
+target_link_libraries(cppnetlib-client-connections ${CMAKE_THREAD_LIBS_INIT})
 if (OPENSSL_FOUND)
     target_link_libraries(cppnetlib-client-connections ${OPENSSL_LIBRARIES})
 endif ()

